packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -229,31 +229,31 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
 
         if (!knownPeerAccount) {
           log(`> wire first_request of ${peerAccount} with outcomeAddress ${peerOutcomeAddress}`);
-          const channel = await this.paymentChannelClient.createChannel(
+          const {channelId} = await this.paymentChannelClient.createChannel(
             this.pseAccount, // seeder
             peerAccount, // leecher
             INITIAL_SEEDER_BALANCE.toString(), // seederBalance,
             INITIAL_LEECHER_BALANCE.toString(), // leecherBalance,
             this.paymentChannelClient.myEthereumSelectedAddress, // seederOutcomeAddress,
             peerOutcomeAddress // leecherOutcomeAddress
           );
-          log(`> created channel with id ${channel.channelId}`);
-          wire.paidStreamingExtension.pseChannelId = channel.channelId;
+          // eslint-disable-next-line require-atomic-updates
+          wire.paidStreamingExtension.pseChannelId = channelId;
 
           this.peersList[torrent.infoHash][peerAccount] = {
             id: peerAccount,
             wire,
             buffer: '0', // (bytes) a value x > 0 would allow a leecher to download x bytes
             beneficiaryBalance: '0', // (wei)
             allowed: false,
-            channelId: channel.channelId
+            channelId
           };
+          log('> BLOCKED: ' + index + ' NEW PEER - Channel Created, id:' + channelId);
           response(false);
-
           this.blockPeer(torrent.infoHash, wire, peerAccount);
         } else if (!knownPeerAccount.allowed || reqPrice.gt(knownPeerAccount.buffer)) {
-          response(false);
```

Comment:
this logs got moved around so that the log messages respect the actual order of execution.